### ISSUE TITLE
export variables cannot be saved

### ISSUE THREAD

**Issue by duane-r** [godotengine/godot#41690]
OS: Archlinux with openbox window manager
Godot: 23ce1dbfd

Add any exported variable to a scene script. Edit it in the scene editor and save the scene -- the value resets to default:

@export var test: int

I may not be using the new gdscript export right. I can't seem to find the documentation.

This is similar to https://github.com/godotengine/godot/issues/41509, but I can't get any type of exported variable to save.

[export-test.zip](https://github.com/godotengine/godot/files/5159015/export-test.zip)

**Comment by akien-mga** [comment]
For future issues, please keep the template sections so that it's easy to parse visually for our contributors (but apart from that you included all the relevant information, so :+1:).

**Comment by akien-mga** [comment]
I can confirm the issue:

`@export var test : int` exposes an int in the inspector with value 0, but after any change, saving resets to `0`. Printing the variable outputs `null` (so it's not an int despite the hint).

`@export var test : int = 1` properly hints it as int with value 1, but similarly changes are discard on save. Printing gives `1`.

#41509 is likely a more specific version of this same GDScript issue indeed.

**Comment by EricEzaM** [comment]
I've had a quick look and it seems like the node property is not returned from `get_property_list` in `ResourceFormatSaverText::save()`, so it is never recognised as a property.

**Comment by Calinou** [comment]
For the record, it works as expected on `tool` scripts (according to #47515).

**Comment by SaracenOne** [comment]
Oh, literally just as filed my own issue for this, it looks like some new commits just dropped in which seem to address, although I think some new issues have been introduced too

**Comment by Blackiris** [comment]
> The current thing I'm seeing from the latest mater ([89637e4](https://github.com/godotengine/godot/commit/89637e45f83a0121bdb1d086a359ce11d981dd8f)), is that this _might_ be resolved, but at the same time, all variables are exposed to the inspector, not just the ones marked as @export

Indeed, I can also confirm it seems fixed but all non-exported variables are now visible in the editor...
The fix was probably from this now merged PR: https://github.com/godotengine/godot/commit/737f09895d24cfd267018bdea2c67c1a4ddd20a6

**Comment by SaracenOne** [comment]
Hmm, I guess it might make sense to close this issue now and open a new one for the newly introduced regression

**Comment by Blackiris** [comment]
This other issue is opened here #48224.
This issue can now be closed.

**Comment by Atermnus** [comment]
Opening this up again with a newly found problem. The exact same issue happens when you use "validate property" to mess around with the inspector to be able to hide/show certain properties.

Just add to minimal reproduction the following lines :

`
@export var SMALL_IMAGE : Texture
@export var MEDIUM_IMAGE : Texture
@export var LARGE_IMAGE: Texture
@export var HUGE_IMAGE : Texture

enum IMAGE_TYPE{
IMAGE_TYPE_1,
IMAGE_TYPE_2,
OTHER
}

@export var TYPE : IMAGE_TYPE 

func _validate_property(property: Dictionary) -> void:
	if property.name == "SMALL_IMAGE":
		if TYPE== IMAGE_TYPE.IMAGE_TYPE_1:
			property.usage = PROPERTY_USAGE_EDITOR
		else:
			property.usage = PROPERTY_USAGE_STORAGE
		
	if property.name == "MEDIUM_IMAGE":
		if TYPE== IMAGE_TYPE.IMAGE_TYPE_2:
			property.usage = PROPERTY_USAGE_EDITOR
		else:
			property.usage = PROPERTY_USAGE_STORAGE
		
	if property.name == "LARGE_IMAGE" or property.name == "HUGE_IMAGE":
		if TYPE== IMAGE_TYPE.OTHER:
			property.usage = PROPERTY_USAGE_EDITOR
		else:
			property.usage = PROPERTY_USAGE_STORAGE
	`

**Comment by AThousandShips** [comment]
Please open a new issue, this was closed over 3 years ago and is unrelated
